src/components/ha-data-table.ts

```diff
@@ -0,0 +1,226 @@
+import {
+  MDCDataTableAdapter,
+  MDCDataTableRowSelectionChangedEventDetail,
+  MDCDataTableFoundation,
+} from "@material/data-table";
+
+import {
+  BaseElement,
+  html,
+  query,
+  queryAll,
+  CSSResult,
+  css,
+  customElement,
+  property,
+  unsafeCSS,
+  classMap,
+} from "@material/mwc-base/base-element.js";
+
+// tslint:disable-next-line
+import style from "@material/data-table/dist/mdc.data-table.min.css";
+
+import "./ha-checkbox";
+// tslint:disable-next-line
+import { HaCheckbox } from "./ha-checkbox";
+
+export interface DataTabelHeaderData {
+  name: string;
+  type?: "numeric";
+  sortable?: boolean;
+  sort?: "desc" | "asc";
+}
+
+@customElement("ha-data-table")
+export class HaDataTable extends BaseElement {
+  @property({ type: Array }) public headers: DataTabelHeaderData[] = [];
+
+  @property({ type: Array }) public data = [];
+
+  @property({ type: Boolean }) public selectable = false;
+
+  protected mdcFoundation!: MDCDataTableFoundation;
+
+  protected readonly mdcFoundationClass = MDCDataTableFoundation;
+
+  @query(".mdc-data-table") protected mdcRoot!: HTMLElement;
```

Comment:
Does this work? I remember trying it once and it didn't work.

Reply:
I does 💃